### PR TITLE
En 8063 fix after rv audit

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
Removed duplicated code and implemented other small optimizations.

### CODE REVIEW

**Review by sasurobert — DISMISSED** [review]

File: vm/systemSmartContracts/delegation.go [hunk 1]
```diff
@@ -1921,7 +1888,7 @@ func (d *delegation) getFund(key []byte) (*Fund, error) {
 }
 
 func (d *delegation) createAndSaveNextFund(address []byte, value *big.Int, fundType uint32) ([]byte, error) {
```
Comment (SebastianMarian) [inline vm/systemSmartContracts/delegation.go:1890]: createAndSaveNexKeyFund

File: vm/systemSmartContracts/delegation.go [hunk 2]
```diff
@@ -1921,7 +1888,7 @@ func (d *delegation) getFund(key []byte) (*Fund, error) {
...
-	fundKey, fund := d.createNextFund(address, value, fundType)
+	fundKey, fund := d.createNextKeyFund(address, value, fundType)
 	err := d.saveFund(fundKey, fund)
```
Comment (SebastianMarian) [inline vm/systemSmartContracts/delegation.go:1892]: saveKeyFund

File: vm/systemSmartContracts/delegation.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply vm/systemSmartContracts/delegation.go:1890]: Done.

File: vm/systemSmartContracts/delegation.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply vm/systemSmartContracts/delegation.go:1892]: Done.

File: vm/systemSmartContracts/delegation.go [hunk 3]
```diff
@@ -1887,9 +1887,9 @@ func (d *delegation) getFund(key []byte) (*Fund, error) {
 	return dFund, nil
 }
 
-func (d *delegation) createAndSaveNextFund(address []byte, value *big.Int, fundType uint32) ([]byte, error) {
+func (d *delegation) createAndSaveNexKeyFund(address []byte, value *big.Int, fundType uint32) ([]byte, error) {
```
Comment (sasurobert) [inline vm/systemSmartContracts/delegation.go:1890]: createAndSaveNextKeyFund ? however KeyFund does not sound so right. and saveKeyFund might be renamed again to saveFund - as it only safe the fund data.

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: vm/systemSmartContracts/delegation.go (hunk 3 above)
Comment (BeniaminDrasovean) [reply vm/systemSmartContracts/delegation.go]: Done.

**Review by SebastianMarian — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**